### ISSUE TITLE
?flush clears Intervention_Manipulations cache files

### ISSUE THREAD

**Issue by christopherdarling** [silverstripe/silverstripe-assets#197]
Running SS 4.3.0-rc1

I have noticed that the first load of pages after deploying, running dev/build?flush=1 pages are incredibly slow on their first load (10-20s! granted, the pages have a lot of large images which are to be resampled). After some debugging, this appears to be down to InterventionImage caches being cleared by ?flush.

Intervention_Manipulations store file dimensions, these are used on every getWidth() / getHeight() call to a Image (including resampled images), this can be quite expensive when a lot of images are included on a page (especially large ones as they're cloned to temp files to extract file size (not quite sure why, perhaps for S3 etc)). IMO Asset filesizes are unlikely to change when a flush is required so not quite sure this is necessary to clear the cache.

**Comment by christopherdarling** [comment]
Still an issue on SS 4.4.3. Looking at `InterventionBackend` it looks as though removing `Flushable` would do the trick but there may be some cases where you would want to empty the cache. It looks as though it stores;
- Dimensions
- Failures

If you don't feel as though it is safe to remove the cache flushing on this class, would you accept a PR with this as a config setting for BC?

**Comment by sb-relaxt-at** [comment]
This has a large performance impact. Having a page with some hundred large images the recreation of the dimensions cache takes some time…

**Comment by dnsl48** [comment]
> If you don't feel as though it is safe to remove the cache flushing on this class, would you accept a PR with this as a config setting for BC?

Yes, please. PRs are welcome.

To make the solution more complete I'd suggest you to consider adding a simple `dev/task` that would clean the cache on-demand (through manual invocation). The task could be as simple as the original `Flushable::flush` implementation of `InterventionBackend`.

**Comment by christopherdarling** [comment]
@dnsl48 PR at #362

**Comment by dnsl48** [comment]
An opt-in disabling of Intervention cache invalidation on flush will be released in `1.6.0` (`Silverstripe CMS 4.6.0`)

**Comment by sheadawson** [comment]
```
SilverStripe\Assets\InterventionBackend:
  flush_enabled: false
```
